src/views/Dashboard/sidebar/index.js

```diff
@@ -41,26 +41,27 @@ export class Sidebar extends Component {
   render() {
     let backdrop;
     const { sideDrawerOpen } = this.state;
-    const { children } = this.props;
+    const { children, profileError, profile } = this.props;
+    const { tokenValid } = this.state;
+
     if (sideDrawerOpen) {
       backdrop = <Backdrop click={this.backdropClickHandler} />;
     }
 
-    if(this.state.tokenValid) {
-      if (this.props.profileError) {        
-        if(this.props.profileError.error) {
-          return <Redirect to="/verify-email"/>;
-        }
-        if(this.props.profileError.message) {
-          localStorage.removeItem('token')
-          return <Redirect to="/signin"/>;
-        }
-      }}
-    if (this.props.profile.data) {
-      if (!this.props.profile.data.managerId) {
-        return <Redirect to="/profile"/>;
-      }
+ 
+    if(tokenValid && profileError && profileError.error) {
+      return <Redirect to="/verify-email"/>;
+    }
+  
+    if(tokenValid && profileError && profileError.message) {
+      localStorage.removeItem('token')
+      return <Redirect to="/signin"/>;
+    }
```

Comment:
```
tokenValid && profileError && profileError.message && ( (localStorage.removeItem('token')) (<Redirect to="/signin"/>))
```

Reply:
The same issue. We can not redirect the user in case,
- token is invalid
- token is malformed
- token is expired

``` jsx
tokenValid && profileError && profileError.message && (localStorage.removeItem('token'), <Redirect to="/signin"/>);
```
However, I think code looks complicate than it should.

<img width="1440" alt="Screen Shot 2020-02-21 at 11 18 00" src="https://user-images.githubusercontent.com/33821734/75020682-e06b1680-549b-11ea-8923-756dfe0fac2a.png">

Reply:
Okey, @NiyoEric , we can leave it that way.